mechanics/input_processing.py

```diff
@@ -1,48 +1,103 @@
-from textblob import TextBlob
-import statistics
-from pathlib import Path
-
-image_folder_path = Path.cwd() / "mechanics" / "common_expressions.json"
-
-
-def text_sentiment_processing(text):
-    blob = TextBlob(text)
-
-    sentiments = []
-
-    for sentence in blob.sentences:
-        sentiments.append(sentence.sentiment.polarity)
-
-    if statistics.fmean(sentiments) > 0.3:
-        return "positive"
+import marvin
+import json
+import os
 
-    if statistics.fmean(sentiments) >= -0.3 and statistics.fmean(sentiments) <= 0.3:
-        return "neutral"
-
-    if statistics.fmean(sentiments) < -0.3:
-        return "negative"
-
-
-def process_message(text):
-    blob = TextBlob(text)
-
-    # with open(str(image_folder_path), "r") as common_text_file:
-    # common_text = json.load(common_text_file)
-
-    for sentence in blob.sentences:
-        for word in sentence.words:
-            print(word.correct())
+# from pydantic import BaseModel
+# from textblob import TextBlob
+from dotenv import load_dotenv
+from pathlib import Path
 
-    return
+load_dotenv()
+
+OAI_KEY = os.getenv("OPEN_AI_API_KEY")
+npc_settings = Path.cwd() / "characters" / "npc" / "npc_settings.json"
+marvin.settings.openai.api_key = OAI_KEY
+
+# class MessagePersonality():
+# "Intelligence": 17,
+#             "Honesty": 97,
+#             "Patience": 35,
+#             "Humor": 66,
+#             "Ambition": 66,
+#             "Affection": 0,
+#             "Confidence": 23,
+#             "Adventurousness": 87,
+#             "Empathy": 25
+
+
+def generate_response(npc_personality: dict, user_input: str) -> str:
+    message_sentiment = marvin.classify(
+        user_input, labels=["friendly", "unfriendly", "neutral"]
+    )
+    message_type = marvin.classify(
+        user_input, labels=["greeting", "farewell", "request", "question", "other"]
+    )
+    top_values = [
+        element[0]
+        for element in (
+            sorted(npc_personality.items(), key=lambda item: item[1], reverse=True)[:2]
+        )
+    ]
+    message_values = json.loads(
+        marvin.cast(
+            user_input,
+            target=str,
+            instructions="Get how much Intelligence, Honesty, Patience, Humor, Ambition, Affection, Confidence, Adventurousness or Empathy the message has; classify in a scale of 0-100",
```

Comment:
so here you tried `MessagePersonality` as a more specialized dict as return value? that should work?